src/components/quiz-detail/QuizDetail.tsx

```diff
@@ -0,0 +1,109 @@
+import React, {useEffect,useState} from 'react'
+import { supabase } from '../../../supabaseClient';
+import {quizCategoriesDetailsType, categoriesType, quizParamType} from '../../types';
+import {convertDateToMonthAndYear} from '../../utils/'
+import { useToast, useQuizPlay, useAuth } from "../../contexts";
+import  { Toaster } from "react-hot-toast";
+import { useNavigate, Link } from 'react-router-dom';
+
+
+const QuizDetail = ({quizid}: quizParamType) => {
+
+  const [quizDetail, setQuizDetail] = useState<quizCategoriesDetailsType|null>();
```

Comment:
Suggestion: you are using "quizDetail" object's property later in the code. Instead of it, you can destructure the object here and then use individual properties like "quiz_name", "quiz_image" etc.